src/Http/Http/src/Internal/DefaultHttpResponse.cs

```diff
@@ -69,7 +69,7 @@ public override Stream Body
             get { return HttpResponseBodyFeature.Stream; }
             set
             {
-                var otherFeature = _features.Collection.Get<IHttpResponseBodyFeature>();
+                var otherFeature = _features.Collection.Get<IHttpResponseBodyFeature>()!;
```

Comment:
💡 In Roslyn, we use a `GetRequired<TFeature>` extension method which calls `Get<TFeature>` and throws a meaningful exception if the required feature is not available. I would recommend using that approach instead of just the `!` operator.

Reply:
Agreed. We've been lax in assuming that the request feature, the response feature, and the response body feature would always be set, and we'd break horribly if they aren't.

Reply:
I filed https://github.com/dotnet/aspnetcore/issues/25471 so we can take this thru API review.